lndclient/router_client.go

```diff
@@ -140,6 +149,22 @@ func (r *routerClient) SendPayment(ctx context.Context,
 	if request.LastHopPubkey != nil {
 		rpcReq.LastHopPubkey = request.LastHopPubkey[:]
 	}
+	if request.KeySend {
+		if !bytes.Equal(request.PaymentHash[:], emptyHash[:]) {
```

Comment:
If payment hash is optional, it would be better to make it type `*lntypes.Hash` to avoid the magic value.

Reply:
agree, much nicer to have the `*lntypes.Hash`, done